### ISSUE TITLE
belongs_to fails in search

### ISSUE THREAD

**Issue by Nerian** [code-and-effect/effective_datatables#116]
Hi,

I have noticed that the search box stopped working on a couple of my tables. The column stops working from effective-datatable version '4.7.7'. Sorting doesn't work either.

This is the code:

```ruby
module ManagerPanel
  class ResourcesDatatable < Effective::Datatable
    datatable do    
      col :contact_id, label: I18n.t('general.contact'), search: { as: :select, collection: current_account.contacts } do |resource|
        contact_link(resource.contact)
      end
    end

    collection do
      current_account.resources
    end

    def current_account
      @account ||= Account.find(attributes[:account_id])
    end
  end
end
```

A `Resource` belongs_to a `Contact`. So I want to show the contact that this resource is assigned to.

It works as intended prior to '4.7.7'. But fails from '4.7.7'.

This is what I see in the log when I perform a search:

Notice that it searches using a string of text (the name), instead of searching using the id. I see that in the Select the `val` for the options is the name of the contact, instead of the ID – prior versions use the ID.
```
COLUMN TOOL: search_column contact_id Johnny Schmeler "resources"."contact_id"
   (4.0ms)  SELECT "contacts"."id" FROM "contacts" WHERE "contacts"."deleted_at" IS NULL AND ("contacts"."stripe_customer_id" ILIKE '%Johnny Schmeler%' OR "contacts"."gocardless_customer_id" ILIKE '%Johnny Schmeler%' OR "contacts"."name" ILIKE '%Johnny Schmeler%' OR "contacts"."email" ILIKE '%Johnny Schmeler%' OR "contacts"."country" ILIKE '%Johnny Schmeler%' OR "contacts"."government_identity_code" ILIKE '%Johnny Schmeler%' OR "contacts"."billing_address" ILIKE '%Johnny Schmeler%' OR "contacts"."phone" ILIKE '%Johnny Schmeler%' OR "contacts"."notes" ILIKE '%Johnny Schmeler%' OR "contacts"."bio" ILIKE '%Johnny Schmeler%' OR "contacts"."tags" ILIKE '%Johnny Schmeler%' OR "contacts"."twitter" ILIKE '%Johnny Schmeler%' OR "contacts"."facebook" ILIKE '%Johnny Schmeler%' OR "contacts"."google_plus" ILIKE '%Johnny Schmeler%' OR "contacts"."youtube" ILIKE '%Johnny Schmeler%' OR "contacts"."instagram" ILIKE '%Johnny Schmeler%' OR "contacts"."pinterest" ILIKE '%Johnny Schmeler%' OR "contacts"."tumblr" ILIKE '%Johnny Schmeler%' OR "contacts"."telegram" ILIKE '%Johnny Schmeler%' OR "contacts"."invite_code" ILIKE '%Johnny Schmeler%')
Completed 500 Internal Server Error in 23ms (Views: 0.3ms | ActiveRecord: 8.0ms)



PG::UndefinedFunction - ERROR:  operator does not exist: character varying[] ~~* unknown
LINE 1: ...o" ILIKE '%Johnny Schmeler%' OR "contacts"."tags" ILIKE '%Jo...
                                                             ^
HINT:  No operator matches the given name and argument types. You might need to add explicit type casts.
:
  app/controllers/application_controller.rb:22:in `block in set_time_zone'
  app/controllers/application_controller.rb:21:in `set_time_zone'
```

Is this a Bug or is there a new way to do what I want?

Thanks!

**Comment by matt-riemer** [comment]
This is a bug. Fixed in 4.7.11. (yay 7-11 version!)

Thanks :)

**Comment by Nerian** [comment]
Thanks for your quick fix! Working like a charm. 👍
